Foundation/NSArray.swift

```diff
@@ -130,7 +130,7 @@ open class NSArray : NSObject, NSCopying, NSMutableCopying, NSSecureCoding, NSCo
 //        }
         let cnt = array.count
         let buffer = UnsafeMutablePointer<AnyObject>.allocate(capacity: cnt)
-        buffer.initialize(from: optionalArray)
+        _ = UnsafeMutableBufferPointer(start: buffer, count: cnt).initialize(from: optionalArray)
```

Comment:
This seems like programming via side-effect. Is there some more explicit way we could express this?